src/runtime/thread_pool_common.h

```diff
@@ -273,24 +276,38 @@ WEAK void worker_thread_already_locked(work *owned_job) {
         if (!job) {
             // There is no runnable job. Go to sleep.
```

Comment:
Can you move the declaration/initialization of spin_count here, so it doesn't need to be explicitly reset?

Reply:
Err, no, you can't...

Reply:
The spin_count should only be reset when the thread actually manages to do something useful. In this implementation, if a thread gets woken up for a job and the work is all already claimed, then spins 40 times waiting for a new job to be enqueued before giving up. If the same thing starts to happen with the next parallel loop, then it doesn't keep spinning - it goes to sleep right away because spin_count is still > 40 from the last job.

Reply:
Note that this makes it different to just adding a spin inside cond_wait. I believe there are never more than 40 spins per work item completed here.